File: GameLibrary/src/cc/lib/zombicide/anims/SlashedAnimation.java
package cc.lib.zombicide.anims;

import cc.lib.game.AGraphics;
import cc.lib.game.AImage;
import cc.lib.game.GRectangle;
import cc.lib.game.Utils;
import cc.lib.zombicide.ZActor;
import cc.lib.zombicide.ZActorAnimation;
import cc.lib.zombicide.ZIcon;

public class SlashedAnimation extends ZActorAnimation {

    final int claws = Utils.randItem(ZIcon.CLAWS.imageIds);
    GRectangle rect;

    public SlashedAnimation(ZActor actor) {
        super(actor, 1000);
        rect = actor.getRect();
    }

    @Override
    protected void draw(AGraphics g, float position, float dt) {
        AImage img = g.getImage(claws);
        g.setTransparencyFilter(1f-position);
        g.drawImage(claws, rect.fit(img));
        g.removeFilter();
    }

    @Override
    protected boolean hidesActor() {
        return false;
    }
}

File: GameLibrary/src/cc/lib/zombicide/anims/SlashedAnimation.kt
package cc.lib.zombicide.anims

import cc.lib.game.AGraphics
import cc.lib.game.GRectangle

import cc.lib.zombicide.ZActor
import cc.lib.zombicide.ZActorAnimation
import cc.lib.zombicide.ZIcon

class SlashedAnimation(actor: ZActor<*>) : ZActorAnimation(actor, 1000) {
	val claws = ZIcon.CLAWS.imageIds.random()
	var r: GRectangle = actor.getRect()
	override fun draw(g: AGraphics, position: Float, dt: Float) {
		val img = g.getImage(claws)
		g.setTransparencyFilter(1f - position)
		g.drawImage(claws, r.fit(img))
		g.removeFilter()
	}

	override fun hidesActor(): Boolean {
		return false
	}

}